core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -66,20 +69,81 @@ public ConstantScoreQueryBuilder boost(float boost) {
         return this;
     }
 
+    /**
+     * @return the boost factor
+     */
+    public float boost() {
+        return this.boost;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("filter");
-        filterBuilder.toXContent(builder, params);
+        queryBuilder.toXContent(builder, params);
+        builder.field("boost", boost);
+        builder.endObject();
+    }
 
-        if (boost != -1) {
-            builder.field("boost", boost);
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        if (queryBuilder == null ) {
+            return null;
         }
-        builder.endObject();
+
+        Query innerFilter = queryBuilder.toQuery(parseContext);
+        if (innerFilter == null ) {
+            return null;
+        }
+
+        Query filter = new ConstantScoreQuery(queryBuilder.toQuery(parseContext));
+        filter.setBoost(boost);
+        return filter;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (queryBuilder == null) {
+            validationException = QueryValidationException.addValidationError("[constant_score] query requires inner null or empty.", validationException);
```

Comment:
clarify the error message specifying what needs to be non null? the inner query...also remove empty, doesnt make sense here